### PR TITLE
DEVDOCS-4484: [update] Stencil, custom templates now support front matter

### PR DESCRIPTION
**Pull request by bc-traciporter**
# [DEVDOCS-4484]

## What changed?
Added warning text in the Specifying custom front matter section.

## Anything else?

[DEVDOCS-4484]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4484?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/stencil-docs/storefront-customization/custom-templates.md [hunk 1]
```diff
@@ -118,7 +118,9 @@ Beyond the single URL mapped to each template in the above examples, you have th
...
+You can't specify front matter on a custom template, but you can still access front matter-injected data. Custom templates inherit front matter from their corresponding default template. Specify the data you would like your custom template to access in the default template's front matter. See [Using Front Matter](/stencil-docs/storefront-customization/using-front-matter) for more information on using front matter. 
+
+The following example shows data you can inject into `templates/pages/brand.html` to access data in `templates/pages/custom/brand/custom-brand.html`. After [mapping](/stencil-docs/storefront-customization/custom-templates#mapping-multiple-urls) the custom template to the default page, the cart object displays on the page.
```
Comment (bc-max) [inline docs/stencil-docs/storefront-customization/custom-templates.md:123]: @bc-traciporter I believe we can remove lines 121 -123 above. Stencil-cli versions `5.2.1+` supports front matter on custom templates. So if a customer defines a custom template with a front matter in his local theme (in `config.stencil.json`), this front matter will be taken into account by stencil-cli. So you don't need to change default template/front matter as we use directly what we have in custom template.

File: docs/stencil-docs/storefront-customization/custom-templates.md [hunk 2]
```diff
@@ -149,6 +151,9 @@ title: Custom template - custom-brand.html
 
 <!-- type: tab-end -->
 
+<!-- theme: warning -->
+> #### Stencil CLI versus storefront
+> When adding front matter directly to a custom template, Stencil CLI behaves as explained above. You must add front matter to the default template. However, when applying a theme to the storefront, it is possible to specify front matter directly to the custom template.
```
Comment (bc-max) [inline docs/stencil-docs/storefront-customization/custom-templates.md:156]: Stencil CLI starting from version `5.2.1` has the same behaviour as a regular mode (eg. live store) - both refular and CLI modes use front matter from custom template, not default one.

**Review by bc-max — CHANGES_REQUESTED** [review]

File: docs/stencil-docs/storefront-customization/custom-templates.md [hunk 3]
```diff
@@ -118,7 +118,9 @@ Beyond the single URL mapped to each template in the above examples, you have th
 
 ## Specifying custom front matter
 
-You can't specify front matter on a custom template, but you can still access front matter-injected data. Custom templates inherit front matter from their corresponding default template. Specify the data you would like your custom template to access in the default template's front matter. See [Using Front Matter](/stencil-docs/storefront-customization/using-front-matter) for more information on using front matter. The following example shows data you can inject into `templates/pages/brand.html` to access data in `templates/pages/custom/brand/custom-brand.html`. After [mapping](/stencil-docs/storefront-customization/custom-templates#mapping-multiple-urls) the custom template to the default page, the cart object displays on the page.
+When adding front matter directly to a custom template, you must add front matter to the default template when using Stencil CLI. However, when applying a theme to the storefront, it is possible to specify front matter directly to the custom template. See [Using Front Matter](/stencil-docs/storefront-customization/using-front-matter) for more information on using front matter.
```
Comment (bc-max) [inline docs/stencil-docs/storefront-customization/custom-templates.md:121]:
```suggestion
You can add front matter directly to a custom template. See [Using Front Matter](/stencil-docs/storefront-customization/using-front-matter) for more information on using front matter. 
```

File: docs/stencil-docs/storefront-customization/custom-templates.md [hunk 4]
```diff
@@ -118,7 +118,9 @@ Beyond the single URL mapped to each template in the above examples, you have th
...
+When adding front matter directly to a custom template, you must add front matter to the default template when using Stencil CLI. However, when applying a theme to the storefront, it is possible to specify front matter directly to the custom template. See [Using Front Matter](/stencil-docs/storefront-customization/using-front-matter) for more information on using front matter. 
+
+The following example shows data you can inject into `templates/pages/brand.html` to access data in `templates/pages/custom/brand/custom-brand.html`. After [mapping](/stencil-docs/storefront-customization/custom-templates#mapping-multiple-urls) the custom template to the default page, the cart object displays on the page.
```
Comment (bc-max) [inline docs/stencil-docs/storefront-customization/custom-templates.md:123]: It looks like you can completely delete line 123 as no injections to default template required.

**Review by bc-max — CHANGES_REQUESTED** [review]

**Review by bc-max — APPROVED** [review]

**Review by stanzikratelbc — DISMISSED** [review]
LGTM. Ready to merge

**Review by stanzikratelbc — DISMISSED** [review]
LGTM. Ready to merge and publish.

**Review by stanzikratelbc — APPROVED** [review]
LGTM. Ready to Merge

**Pull request merged**